apps/x509.c

```diff
@@ -1097,3 +1105,93 @@ static int purpose_print(BIO *bio, X509 *cert, X509_PURPOSE *pt)
     }
     return 1;
 }
+
+static int parse_ext_names(char *names, const char **result)
+{
+    char *p, *q;
+    int cnt = 0, idx = 0, len = 0;
+
+    p = q = names;
+    len = strlen(names) + 1;
+
+    while (q - names <= len) {
```

Comment:
Simply use `strchr` in while condition will require extra code to deal the 'last' name, e.g. 'c' of "a,b,c", something would be like:

```
    while ((q = strchr(p, ',')) != NULL) {
        if (p != q) {
            /* found */
            if (result != NULL) {
                result[idx++] = p;
                *q = '\0';
            }
            cnt++;
        }
        p = q + 1;
    }

    if (*p != '\0') {
        if (result != NULL)
            result[idx++] = p;
        cnt++;
    }
```
And it doesn't seem 'neat' since duplicated code introduced. If merge the extra 'if' into while, the whole while loop seems also not that simple as the original one.

But there is indeed a bug I found in the original logic, `len = strlen(names) + 1;` shoudl be `len = strlen(names);` to avoid pointer referenced out-of-scope.

> Is it worth considering a check for an empty name

This is included in the original logic, if `names = "\0"` is passed, `cnt = 0` will be returned and the caller will handle this case.